synapse/app/federation_sender.py

```diff
@@ -262,6 +262,8 @@ def _on_new_receipts(self, rows):
             # we only want to send on receipts for our own users
             if not self._is_mine_id(receipt.user_id):
                 continue
+            if receipt.data.get("hidden", False):
+                return  # do not send over federation
```

Comment:
we don't want to skip *all* receipts in the list, iirc:

```suggestion
                continue  # do not send over federation
```